packages/common/src/visualizations/types/index.ts

```diff
@@ -95,9 +96,7 @@ export type VizTableDisplay = {
     // On vis column config, visible, label and frozen, at least seem like display options
 };
 
-export type PivotIndexColum =
-    | { reference: string; type: VizIndexType }
-    | undefined;
+export type PivotIndexColum = { reference: string; type: VizIndexType };
```

Comment:
Don't use `undefined` here. Let the parent object define `PivotIndexColum | undefined` instead.